File: openmp/testsuite/fortran/omp_task_shared.f
<ompts:test>
<ompts:testdescription> Test to see if implied shared works correctly</ompts:testdescription>
<ompts:ompversion>3.0</ompts:ompversion>
<ompts:directive>omp task</ompts:directive>
<ompts:dependences>omp single, omp task firstprivate</ompts:dependences>
<ompts:testcode>
      INCLUDE "omp_my_sleep.f"

      INTEGER FUNCTION <ompts:testcode:functionname>omp_task_shared</ompts:testcode:functionname>()
        IMPLICIT NONE
        INCLUDE "omp_testsuite.f"
        <ompts:orphan:vars>
        external my_sleep
        INTEGER i
        COMMON /orphvars/ i
        </ompts:orphan:vars>
        INTEGER rslt
        INTEGER k

        i = 0
        k = 0
        rslt = 0

!$omp parallel private(k) shared(i)
!$omp single
        do k=1, NUM_TASKS
        <ompts:orphan>
!$omp task <ompts:crosscheck>firstprivate(i)</ompts:crosscheck>
!$omp+     <ompts:check>shared(i)</ompts:check>
!$omp atomic
            i = i + 1
!$omp end task
        </ompts:orphan>
        end do
!$omp end single
!$omp end parallel

        rslt = i
        if (rslt .eq. NUM_TASKS) then
            <testfunctionname></testfunctionname> = 1
        else
            <testfunctionname></testfunctionname> = 0
        end if

      END FUNCTION
</ompts:testcode>
</ompts:test>

File: openmp/testsuite/c/omp_task_shared.c
<ompts:test>
<ompts:testdescription> Test to see if implied shared works correctly</ompts:testdescription>
<ompts:ompversion>3.0</ompts:ompversion>
<ompts:directive>omp task</ompts:directive>
<ompts:dependences>omp single, omp task firstprivate</ompts:dependences>
<ompts:testcode>
#include <stdio.h>
#include <math.h>
#include "omp_testsuite.h"




/* Utility function do spend some time in a loop */
int <ompts:testcode:functionname>omp_task_imp_shared</ompts:testcode:functionname> (FILE * logFile)
{
   <ompts:orphan:vars>
    int i;
   </ompts:orphan:vars>
    i=0;
    int k = 0;
    int result = 0;

   #pragma omp parallel
    {
       #pragma omp single
          for (k = 0; k < NUM_TASKS; k++)           
                  {
                    <ompts:orphan>
                    #pragma omp task <ompts:crosscheck> firstprivate(i) </ompts:crosscheck> <ompts:check> shared(i)</ompts:check>
                        {
                          #pragma omp atomic
                            i++;
                          //this should be shared implicitly
                                
                        }
                  </ompts:orphan> 
                  }

    }

result = i;
return ((result == NUM_TASKS));
     
}
</ompts:testcode>
</ompts:test>
